src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -81,6 +81,13 @@ private void AddWellKnownTypes(IRootingServiceProvider rootProvider)
             {
                 rootProvider.AddCompilationRoot(stringType, "String type is always generated");
             }
+
+            var coreLib = _typeSystemContext.GetModuleForSimpleName("System.Private.Corelib");
+
+            // Adding for CPP codeGen
+            var eeTypeType = coreLib.GetKnownType("Internal.Runtime", "EEType");
+            rootProvider.AddCompilationRoot(eeTypeType, "Enables C++ Code Generation");
```

Comment:
The classlib code refers to EETypes specifically, so to avoid regression, I've continued using the EEType. Would a tweak in the IL importer be a better choice? (i.e. change all `EEType* ` to `void*`)